logical/framework/backend.go

```diff
@@ -280,6 +280,11 @@ func (b *Backend) Type() logical.BackendType {
 	return b.BackendType
 }
 
+// RegisterLicense performs backend license registration
+func (b *Backend) RegisterLicense(interface{}) error {
```

Comment:
Given that nearly all plugins use *framework.Backend this would appear to essentially make the functionality never work.

I think the right way to deal with this is to have a framework.Backend data member that is a function with the same signature called e.g. LicenseRegistrationFunc. If a backend has such a function to handle registrations, it can set that to an appropriate function; if not it remains nil. Then this function checks whether that variable is nil or not and does the right thing.